core/metrics/src/pusher.rs

```diff
@@ -111,57 +118,127 @@ async fn try_get_default_id() -> anyhow::Result<IdentityInfo> {
                 .unwrap_or_else(|| id.node_id.to_string())
         ))?;
+
+    for (label, value) in &ctx.labels {
+        url.path_segments_mut()
+            .map_err(|_e| anyhow::anyhow!("Couldn't add label `{label}`"))?
+            .push(label)
+            .push(value);
+    }
+
     Ok(String::from(url.as_str()))
 }
 
 #[cfg(test)]
 mod test {
     use crate::pusher::get_push_url;
+    use std::collections::HashMap;
+
     use ya_core_model::identity::IdentityInfo;
+    use ya_service_api::MetricsCtx;
+
+    fn default_id_info() -> IdentityInfo {
+        IdentityInfo {
+            alias: Some("node1".into()),
+            node_id: Default::default(),
+            is_locked: false,
+            is_default: false,
+            deleted: false,
+        }
+    }
+
+    fn labels(labels: &[(&str, &str)]) -> HashMap<String, String> {
+        labels
+            .iter()
+            .map(|(k, v)| (k.to_string(), v.to_string()))
+            .collect()
+    }
 
     #[test]
     fn test_get_push_url_with_slashes() {
+        let ctx = MetricsCtx {
+            job: "community.1".into(),
+            ..MetricsCtx::default()
+        };
         let url = get_push_url(
             "http://a",
             &IdentityInfo {
                 alias: Some("ala/ma/kota".into()),
-                node_id: Default::default(),
-                is_locked: false,
-                is_default: false,
-                deleted: false,
+                ..default_id_info()
             },
-            "community.1",
+            &ctx,
         )
         .unwrap();
         assert_eq!("http://a/metrics/job/community.1/instance/0x0000000000000000000000000000000000000000/hostname/ala%2Fma%2Fkota", url);
     }
 
     #[test]
     fn test_get_push_url_with_pletters() {
+        let ctx = MetricsCtx {
+            job: "community.1".into(),
+            ..MetricsCtx::default()
+        };
         let url = get_push_url(
             "http://a",
             &IdentityInfo {
                 alias: Some("zażółć?gęślą!jaźń=".into()),
-                node_id: Default::default(),
-                is_locked: false,
-                is_default: false,
-                deleted: false,
+                ..default_id_info()
             },
-            "community.1",
+            &ctx,
         )
         .unwrap();
         assert_eq!("http://a/metrics/job/community.1/instance/0x0000000000000000000000000000000000000000/hostname/za%C5%BC%C3%B3%C5%82%C4%87%3Fg%C4%99%C5%9Bl%C4%85%21ja%C5%BA%C5%84%3D", url);
     }
+
+    #[test]
+    fn test_get_push_url_with_single_label() {
+        let ctx = MetricsCtx {
+            job: "community.1".into(),
+            labels: labels(&[("group", "random")]),
+            ..MetricsCtx::default()
+        };
+        let url = get_push_url("http://a", &default_id_info(), &ctx).unwrap();
+        assert_eq!("http://a/metrics/job/community.1/instance/0x0000000000000000000000000000000000000000/hostname/node1/group/random", url);
```

Comment:
I assume this `community.1` is a reason you remove some chars here 
https://github.com/golemfactory/yagna/pull/3155/files#diff-28888f6916f83521f834cc9d693aa6412458f751628bcb02dda30600ef5e38c5R12

Reply:
Community was working correctly earlier and wasn't checked at all. But I wanted to have ability to use labels with special characters and they were looking really ugly in Grafana